internal/gcsx/random_reader.go

```diff
@@ -372,6 +372,7 @@ func (rr *randomReader) startRead(
 				Start: uint64(start),
 				Limit: uint64(end),
 			},
+			ReadCompressed: rr.object.ContentEncoding == "gzip",
```

Comment:
Usually, in general, content encoding can have multiple values in order in which they were applied https://datatracker.ietf.org/doc/html/rfc7231#section-3.1.2.2. Did we consider that case in this PR ?

Reply:
It was mentioned in the approved design-doc (http://shortn/_3RP5eBmjnm) that this change only considers content-encoding `gzip` as compressed and treats everything else the same.

Reply:
Understood, I meant is it possible that an object has the content-encoding has value: X, gzip which means it is first encoded by X and then gzipped ? If yes, then should we read this object decompressed 
?

Reply:
> I meant is it possible that an object has the content-encoding has value: X, gzip which means it is first encoded by X and then gzipped

Okay, I wasn't aware that such a thing existed.

> If yes, then should we read this object decompressed ?

The decision, as captured in the design doc, is to always return the compressed data in case of content-encoding `gzip` and the normal data in all other cases. This in other words means returning the data to the end user exactly as it is stored on the GCS. 
The above flag is only to pass the flag to go-storage-library to avoid trying to read the object as decompressed for objects with content-encoding `gzip`.

Reply:
@sethiay ^